### PR TITLE
add extended test for jenkins bc with env vars

### PR DESCRIPTION
**Pull request by gabemontero**
@openshift/devex PTAL

### CODE REVIEW

**Comment by gabemontero** [comment]
[testextended][extended:core("Pipeline with env vars")]

File: test/extended/testdata/samplepipeline-withenvs.yaml [hunk 1]
```diff
@@ -0,0 +1,332 @@
+apiVersion: v1
```
Comment (bparees) [inline test/extended/testdata/samplepipeline-withenvs.yaml]: you should be able to strip a bunch of stuff out of this sample:

no need to have DB service/deployment
no need to have the app service/deployment (you just need the build)

File: test/extended/testdata/samplepipeline-withenvs.yaml [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+      jenkinsPipelineStrategy:
+        env:
+          - name: FOO1
+            value: BAR1
+        jenkinsfile: |-
+          node('nodejs') {
+            stage('build') {
+              echo "FOO1 is ${env.FOO1}"
+              echo "FOO2 is ${env.FOO2}"
+              openshiftBuild(buildConfig: '${NAME}', showBuildLogs: 'true')
```
Comment (bparees) [inline test/extended/testdata/samplepipeline-withenvs.yaml]: for that matter i don't think you even need this buildconfig/build.

File: test/extended/builds/pipeline.go [hunk 3]
```diff
@@ -177,6 +178,55 @@ var _ = g.Describe("[builds][Slow] openshift pipeline build", func() {
...
+			debugAnyJenkinsFailure(br, oc.Namespace()+"-sample-pipeline-withenvs", oc, true)
+			br.AssertSuccess()
+
+			g.By("get build console logs and see if succeeded")
+			_, err = j.WaitForContent("Finished: SUCCESS", 200, 10*time.Minute, "job/%s-sample-pipeline-withenvs/lastBuild/consoleText", oc.Namespace())
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("get build console logs and see if env is set")
+			_, err = j.WaitForContent("FOO2 is BAR2", 200, 10*time.Minute, "job/%s-sample-pipeline-withenvs/lastBuild/consoleText", oc.Namespace())
+			o.Expect(err).NotTo(o.HaveOccurred())
```
Comment (bparees) [inline test/extended/builds/pipeline.go]: should run a third build w/ no env args and confirm that FOO2 is back to null(? emptystring?).  (or just reverse the order of the existing 2 tests should be sufficient).

File: test/extended/testdata/samplepipeline-withenvs.yaml (hunk 1 above)
Comment (gabemontero) [inline test/extended/testdata/samplepipeline-withenvs.yaml]: part of next push

File: test/extended/testdata/samplepipeline-withenvs.yaml (hunk 2 above)
Comment (gabemontero) [inline test/extended/testdata/samplepipeline-withenvs.yaml]: part of next push

File: test/extended/builds/pipeline.go (hunk 3 above)
Comment (gabemontero) [inline test/extended/builds/pipeline.go]: the latter is part of the next push

**Comment by gabemontero** [comment]
updates pushed

**Comment by bparees** [comment]
lgtm, will merge on test pass

**Comment by gabemontero** [comment]
travis failure - forgot to update bindata after yaml change

**Comment by gabemontero** [comment]
On the extended test run, the new ci tool based approach doesn't seem to deal with ginkgo focus like the old vagrant approach does.

It did not think there were any tests to run:

```
[ERROR] No tests would be run
```

I'm going to try and initiate the run at a higher level focus and see what happens.

**Comment by bparees** [comment]
@gabemontero I think the issue is your quotes.

[testextended][extended:core(Pipeline with env vars)]

**Comment by gabemontero** [comment]
[testextended][extended:core("openshift pipeline build")]

**Comment by gabemontero** [comment]
ah - gotcha ... just deleted my repost

**Comment by gabemontero** [comment]
the old vagrant style tolerated quotes

**Comment by gabemontero** [comment]
looks like I didn't delete in time and it captured both focuses ... this should be interesting

**Comment by bparees** [comment]
> the old vagrant style tolerated quotes

that's an issue for @stevekuznetsov

**Comment by stevekuznetsov** [comment]
On a scale of 1 to 10.... how bad do you want quotes

**Comment by gabemontero** [comment]
@stevekuznetsov as long as i know not to use them, i'll rank it a 1 (1 == less important)

**Comment by gabemontero** [comment]
The system did not handle the dual comments from @bparees and I very well:

```
+ make test-extended 'SUITE=core("openshift pipeline build"),core' 'FOCUS=Pipeline with env vars'
test/extended/core("openshift pipeline build"),core.sh --ginkgo.focus="Pipeline with env vars"
/bin/sh: -c: line 0: syntax error near unexpected token `"openshift pipeline build"'
/bin/sh: -c: line 0: `test/extended/core("openshift pipeline build"),core.sh --ginkgo.focus="Pipeline with env vars"'
make: *** [test-extended] Error 1
++ export status=FAILURE
++ status=FAILURE
```

I'll clean up and repost.

**Comment by gabemontero** [comment]
[testextended][extended:core(Pipeline with env vars)]

**Comment by stevekuznetsov** [comment]
Something doesn't seem right but I'm not sure at what level this is broken...


```
+ make test-extended SUITE=core 'FOCUS=Pipeline with env vars'
test/extended/core.sh --ginkgo.focus="Pipeline with env vars"
...
[INFO] Running custom: --ginkgo.focus=Pipeline with env vars
...
[ERROR] No tests would be run
```

**Comment by gabemontero** [comment]
Still says not tests to run:  `[ERROR] No tests would be run`

Going to try the broader scope without quotes.

**Comment by gabemontero** [comment]
[testextended][extended:core(openshift pipeline build)]

**Comment by gabemontero** [comment]
Still with the no tests would be run:

```
[INFO] Running custom: --ginkgo.focus=openshift pipeline build
I0323 13:39:44.820931    2955 plugins.go:74] Registered admission plugin "LimitRanger"
I0323 13:39:44.820956    2955 plugins.go:74] Registered admission plugin "openshift.io/ImageLimitRange"

[ERROR] No tests would be run

```

@stevekuznetsov - I think we have a problem with specifying ginkgo focuses with the new ci tooling.

**Comment by stevekuznetsov** [comment]
That part is the same as it always was before and seems to work fine for the image and build jobs ... can you try running it locally?

```
make test-extended SUITE=core FOCUS='openshift pipeline build'
```

**Comment by gabemontero** [comment]
Good call - running it locally like that fails in the same way:

```
[INFO] Running custom: --ginkgo.focus=openshift pipeline build
I0323 14:12:59.938034   21506 plugins.go:74] Registered admission plugin "LimitRanger"
I0323 14:12:59.938054   21506 plugins.go:74] Registered admission plugin "openshift.io/ImageLimitRange"
[ERROR] No tests would be run
```

And in fact running `test/extended/core.sh --ginkgo.focus=...` now also fails for me locally (which is what I typically use).

I've rebased a few times today, so in theory a recent change introduced this (though at first blush not sure what to git blame on, except perhaps this plugins.go file).

I'll start poking around, but of course any insights are welcome.

**Comment by stevekuznetsov** [comment]
All I know is `--ginkgo.focus` is treated as a regex, but I would have expected this to work.

**Comment by gabemontero** [comment]
I might have an inkling on what is breaking (not necessarily what changed to cause the breakage) ... working on a workaround to confirm the theory.

**Comment by gabemontero** [comment]
Actually, based on when the following commit merged, it might be this from git-blame:

```
42deef1ff (Clayton Coleman         2017-03-12 15:57:33 -0400 1215) 	switch {
42deef1ff (Clayton Coleman         2017-03-12 15:57:33 -0400 1216) 	case len(elem) == 0:
42deef1ff (Clayton Coleman         2017-03-12 15:57:33 -0400 1217) 		panic("must specify path")
42deef1ff (Clayton Coleman         2017-03-12 15:57:33 -0400 1218) 	case len(elem) > 3 && elem[0] == ".." && elem[1] == ".." && elem[2] == "examples":
42deef1ff (Clayton Coleman         2017-03-12 15:57:33 -0400 1219) 		elem = elem[2:]
42deef1ff (Clayton Coleman         2017-03-12 15:57:33 -0400 1220) 	case len(elem) > 3 && elem[0] == ".." && elem[1] == "integration":
42deef1ff (Clayton Coleman         2017-03-12 15:57:33 -0400 1221) 		elem = append([]string{"test"}, elem[1:]...)
42deef1ff (Clayton Coleman         2017-03-12 15:57:33 -0400 1222) 	case elem[0] == "testdata":
42deef1ff (Clayton Coleman         2017-03-12 15:57:33 -0400 1223) 		elem = append([]string{"test", "extended"}, elem...)
42deef1ff (Clayton Coleman         2017-03-12 15:57:33 -0400 1224) 	default:
42deef1ff (Clayton Coleman         2017-03-12 15:57:33 -0400 1225) 		panic(fmt.Sprintf("Fixtures must be in test/extended/testdata or examples not %s", path.Join(elem...)))
42deef1ff (Clayton Coleman         2017-03-12 15:57:33 -0400 1226) 	}
```

When I disable the use of `os::test::extended::test_list` prior to calling `extended.test "$@"`, I'm getting the panic on line 225, where it posts this message:

`Fixtures must be in test/extended/testdata or examples not ../../test/extended/testdata/samplepipeline-withenvs.yaml`

**Comment by gabemontero** [comment]
ding ding ding ... we have a winner ... can execute locally again.  I'll push an update in a moment to adjust for the changes in framework.go I noted above.

**Comment by gabemontero** [comment]
OK the tests passed (all of openshift pipeline build, including the new pipeline with env vars).

I believe it is ready for your merge comment @bparees when convenient for you

**Comment by gabemontero** [comment]
it looks like jenkins is about to shutdown though

**Comment by bparees** [comment]
[merge]

**Pull request merged**